File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-2.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        unordered_map <char, int> a;
        int l = 0;
        int res = 0;
        for (int i = 0; i < s.size(); i++) {
            if (a.count(s[i]) == 1) {
                l = max(l, a[s[i]]+1);
            }
            a[s[i]]=i;
            res = max(res, i - l + 1);
        }
        return res;
    }
};

File: Data Structures & Algorithms/meeting-schedule-ii/submission-2.py
class Solution:
    def minMeetingRooms(self, intervals: List[Interval]) -> int:
        if not intervals:
            return 0
        
        starts = sorted([i.start for i in intervals])
        ends = sorted([i.end for i in intervals])
        
        s_ptr = 0
        e_ptr = 0
        res = 0
        count = 0
        while s_ptr < len(starts):
            if starts[s_ptr] < ends[e_ptr]:
                count += 1
                s_ptr += 1
            else:
                count -= 1
                e_ptr += 1
            res = max(res, count)
            
        return res